client/keys/migrate.go

```diff
@@ -22,76 +24,111 @@ const migratePassphrase = "NOOP_PASSPHRASE"
-		Long: `Migrate keys from the legacy on-disk secret store to the OS keyring.
-The command asks for every passphrase. If the passphrase is incorrect, it skips the respective key.
+		Short: "Migrate keys from the legacy (db-based) Keybase",
+		Long: `Migrate key information from the legacy (db-based) Keybase to the new keyring-based Keybase.
+For each key material entry, the command will prompt if the key should be skipped or not. If the key
+is not to be skipped, the passphrase must be entered. The key will only be migrated if the passphrase
+is correct. Otherwise, the command will exit and migration must be repeated.
+
+It is recommended to run in 'dry-run' mode first to verify all key migration material.
 `,
 		Args: cobra.ExactArgs(0),
 		RunE: runMigrateCmd,
 	}
 
-	cmd.Flags().Bool(flags.FlagDryRun, false, "Do everything which is supposed to be done, but don't write any changes to the keyring.")
+	cmd.Flags().Bool(flags.FlagDryRun, false, "Run migration without actually persisting any changes to the new Keybase")
 	return cmd
 }
 
 func runMigrateCmd(cmd *cobra.Command, args []string) error {
 	// instantiate legacy keybase
 	rootDir := viper.GetString(flags.FlagHome)
-	legacykb, err := NewKeyBaseFromDir(rootDir)
+	legacyKb, err := NewKeyBaseFromDir(rootDir)
 	if err != nil {
 		return err
 	}
 
 	// fetch list of keys from legacy keybase
-	oldKeys, err := legacykb.List()
+	oldKeys, err := legacyKb.List()
 	if err != nil {
 		return err
 	}
 
-	// instantiate keyring
-	var keyring keys.Keybase
 	buf := bufio.NewReader(cmd.InOrStdin())
 	keyringServiceName := types.GetConfig().GetKeyringServiceName()
+
+	var (
+		tmpDir  string
+		keybase keys.Keybase
+	)
+
 	if viper.GetBool(flags.FlagDryRun) {
-		keyring, err = keys.NewTestKeyring(keyringServiceName, rootDir)
+		tmpDir, err = ioutil.TempDir("", "keybase-migrate-dryrun")
+		if err != nil {
+			return errors.Wrap(err, "failed to create temporary directory for dryrun migration")
+		}
+
+		defer os.RemoveAll(tmpDir)
+
+		keybase, err = keys.NewTestKeyring(keyringServiceName, tmpDir)
 	} else {
-		keyring, err = keys.NewKeyring(keyringServiceName, rootDir, buf)
+		keybase, err = keys.NewKeyring(keyringServiceName, rootDir, buf)
 	}
 	if err != nil {
 		return errors.Wrap(err, fmt.Sprintf(
-			"failed to initialize keyring for service %s at directory %s",
-			keyringServiceName, rootDir))
+			"failed to initialize keybase for service %s at directory %s",
+			keyringServiceName, rootDir,
+		))
 	}
 
 	for _, key := range oldKeys {
-		legKeyInfo, err := legacykb.Export(key.GetName())
+		legKeyInfo, err := legacyKb.Export(key.GetName())
 		if err != nil {
 			return err
 		}
 
 		keyName := key.GetName()
 		keyType := key.GetType()
-		cmd.PrintErrf("Migrating %s (%s) ...\n", key.GetName(), keyType)
+
+		// skip key if already migrated
+		if _, err := keybase.Get(keyName); err == nil {
```

Comment:
double-checking here, this returns `err = nil` when already migrated?

Reply:
Yeah, it returns `nil` when the key already exists (i.e already migrated or manually added).